src/Nethermind/Nethermind.Consensus.AuRa/Validators/ContractBasedValidator.cs

```diff
@@ -134,7 +134,7 @@ public override void OnBlockProcessingStart(Block block, ProcessingOptions optio
                     bool loadedValidatorsAreSameInStore = (ValidatorStore.GetValidators()?.SequenceEqual(Validators) == true);
                     if (!loadedValidatorsAreSameInStore)
                     {
-                        ValidatorStore.SetValidators(block.Number - 1, Validators);
+                        ValidatorStore.SetValidators(_blockFinalizationManager.GetLastLevelFinalizedBy(block.ParentHash), Validators);
```

Comment:
It is a change without a test proving that the previous thing was wrong and the new one is correct.